kx/kx.go

```diff
@@ -0,0 +1,104 @@
+package kx
+
+import (
+	"crypto/rand"
+	"errors"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/curve25519"
+)
+
+const SeedBytes = 32
+const SecretKeyBytes = 32
+const PublicKeyBytes = 32
+const SessionKeyBytes = 32
+const scalarMultBytes = 32
+
+var invalidSeed = errors.New("invalid seed")
+var cryptoError = errors.New("crypto error")
+var notImplemented = errors.New("not implemented")
+
+type KeyPair struct {
+	pk []byte
+	sk []byte
+}
+
+func NewKeyPair(seed []byte) (*KeyPair, error) {
+	var err error
+	if seed != nil {
+		if len(seed) < SeedBytes {
+			return nil, invalidSeed
+		}
+	} else {
+		seed = make([]byte, SeedBytes)
+		_, err = rand.Read(seed)
+		if err != nil  {
+			return nil, err
+		}
+	}
+
+	kp := new(KeyPair)
+
+	hash, _ := blake2b.New(SecretKeyBytes, nil)
+	hash.Write(seed)
+	kp.sk = hash.Sum(nil)
+
+	if len(kp.sk) != SecretKeyBytes {
+		return nil, cryptoError
+	}
+
+	kp.pk, err = curve25519.X25519(kp.sk, curve25519.Basepoint)
```

Comment:
why not let this be passed into the function?

Reply:
can you elaborate? what this and what function?

Reply:
curve25519.X25519 that function. i was asking it i conforms to some interface so that one could change/not use curve25519?

Reply:
no, and we cannot _not use_ curve25519 because libsodium is using that, key exchange won't work with anything else